auth.ts

```diff
@@ -1,27 +1,34 @@
-import NextAuth from "next-auth"
-import type { Provider } from "next-auth/providers"
-import Google from "next-auth/providers/google"
-import Discord from "next-auth/providers/discord"
+import NextAuth from "next-auth";
+import type { Provider } from "next-auth/providers";
+import Google from "next-auth/providers/google";
+import Discord from "next-auth/providers/discord";
+import { createUser } from "./server-actions/user-actions";
 
-const providers: Provider[] = [
-  Google,
-  Discord,
-]
+const providers: Provider[] = [Google, Discord];
 
 export const providerMap = providers
   .map((provider) => {
     if (typeof provider === "function") {
-      const providerData = provider()
-      return { id: providerData.id, name: providerData.name }
+      const providerData = provider();
+      return { id: providerData.id, name: providerData.name };
     } else {
-      return { id: provider.id, name: provider.name }
+      return { id: provider.id, name: provider.name };
     }
   })
-  .filter((provider) => provider.id !== "credentials")
+  .filter((provider) => provider.id !== "credentials");
 
 export const { handlers, auth, signIn, signOut } = NextAuth({
   providers,
   pages: {
     signIn: "/signin",
   },
-})
\ No newline at end of file
+  events: {
+    async signIn(user) {
```

Comment:
`user.user` looks odd. I followed the api's naming convention in my implementation `message` I think we should return to that implementation.

Reply:
![image](https://github.com/user-attachments/assets/a2aa8fce-a725-4786-b46d-9961a36a5c03)
https://next-auth.js.org/v3/adapters/models#user - `user.user` looks weird, but it works.

I needed a way to write to the DB in NextAuth. Instead of using callbacks, we can use [events](https://next-auth.js.org/configuration/events). With events, they only execute the async function after a successful sign-in.
This implementation successfully writes users to the database after a successful sign-in.